src/serial.rs

```diff
@@ -1,3 +1,41 @@
+//! # Serial Communication
+//! This module contains the functions to utilize the USART (Universal
+//! synchronous asynchronous receiver transmitter (USART)
+//!
+//!
+//! ## Example usage:
+//! ```rust
+//!// prelude: create handles to the peripherals and registers
```

Comment:
I would add at least one space between //! and the code for reading ease when reading the documentation un-rendered.